src/math_functions.cpp

```diff
@@ -692,25 +692,17 @@ double maxwell_spectrum(double T, uint64_t* seed) {
 
 
 double normal_variate(double mean, double standard_deviation, uint64_t* seed) {
-  // perhaps there should be a limit to the number of resamples
-  while ( true ) {
-    double v1 = 2 * prn(seed) - 1.;
-    double v2 = 2 * prn(seed) - 1.;
-
-    double r = std::pow(v1, 2) + std::pow(v2, 2);
-    double r2 = std::pow(r, 2);
-    if (r2 < 1) {
-      double z = std::sqrt(-2.0 * std::log(r2)/r2);
-      z *= (prn(seed) <= 0.5) ? v1 : v2;
-      return mean + standard_deviation*z;
-    }
-  }
+  // now correct method to sample a normal distubted number
+  double v1 = 2 * prn(seed) - 1.;
+  double v2 = 2 * prn(seed) - 1.;
+  double r = std::pow(v1, 2) + std::pow(v2, 2);
+  if ( r == 0 || r > 1 ) return normal_variate(mean,standard_deviation,seed);
+  double z = std::sqrt(-2.0 * std::log(r)/r);
+  return mean + standard_deviation*z*v1;
```

Comment:
Not a fan of the recursion. How about the following:
```suggestion
  // Sample a normal variate using Marsaglia's polar method
  double x, y, r2;
  do {
    x = 2 * prn(seed) - 1.;
    y = 2 * prn(seed) - 1.;
    r2 = x*x + y*y;
  } while (r2 > 1 || r2 == 0);
  double z = x * std::sqrt(-2.0 * std::log(r2)/r2);
  return mean + standard_deviation*z;
```